shared/api/cluster.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+// Cluster represents high-level information about a LXD cluster.
+type Cluster struct {
+	Name string `json:"name" yaml:"name"`
+}
```

Comment:
I'd have expected this to look like:
```
type Cluster struct {
    // True if clustering is enabled
    Enabled bool `json:"enabled" yaml:"enabled"`

    // Same as Name but less likely to be confused for the cluster name (which we don't have)
    ServerName string `json:"server_name" yaml:"server_name"`

    // Global cluster status (rather than member specific), should be "Running" or "Broken" pretty much, "Inactive" or whatever our closest status is when disabled.
    Status string `json:"status" yaml:"status"`
}

Reply:
I can surely rename "Name" to "ServerName".

Re "Enabled", I mentioned that in comment above: do we really want it? As it's essentially redundant with ServerName. Currently, if ServerName != "", then Enabled == true, otherwise Enabled == false. I don't see a case were this correlation would not apply.

Re "Status", I can add it, but I'm not entirely sure what the semantics of it would be yet, that's why I'd suggest we hold it back for now and add it later.

Reply:
I added the Enabled field, as requested on IRC.